share/completions/zstd.fish

```diff
@@ -0,0 +1,58 @@
+# Completions for zstd
+
+complete -c zstd -s z -l compress -d "Compress (default)"
+complete -c zstd -s d -l decompress -l uncompress -d "Decompress" -x -a "
+(
+    __fish_complete_suffix .zst
+)
+"
+
+complete -c zstd -s t -l test -d "Test the integrity"
+complete -r -c zstd -l train -d "Create a dictionary from specified file(s)"
+complete -c zstd -s l -l list -d "List information about .zst file(s)"
+
+for level in (seq 1 19)
+    complete -c zstd -o $level -d "Set compression level"
+end
+
+complete -c zstd -l fast -d "Ultra-fast compression"
+complete -c zstd -l ultra -d "Enable compression level beyond 19"
+complete -c zstd -l long -d "Enable long distance matching with specified windowLog"
+
+for threads in (seq 0 (count (cat /proc/cpuinfo | string match -r "processor")))
```

Comment:
This could get noisy when running on a machine with many CPUs. Maybe it's better to offer only two options: `-T0` to use all available CPUs, and `-T#` where the user needs to type the desired number.
I think `/proc/cpuinfo` doesn't exist on Mac OS, so this would print an error when loading the completions.
Also nit: `string match -r "processor" < /proc/cpuinfo` to get rid of `cat`.

Reply:
>I think /proc/cpuinfo doesn't exist on Mac OS

Also OpenBSD, and FreeBSD by default.

I'd also just drop this, possibly in favor of `seq 0 8` (i.e. just "some numbers") and the auto-detection one.